services/motion/explore/explore.go

```diff
@@ -372,18 +402,13 @@ func (ms *explore) checkForObstacles(
 // executePlan will carry out the desired motionplan plan.
 func (ms *explore) executePlan(ctx context.Context, kb kinematicbase.KinematicBase, plan motionplan.Plan) {
 	// Iterate through motionplan plan
-
-	ms.planMutex.Lock()
-	ms.planStep = 0
-	ms.planMutex.Unlock()
```

Comment:
curious what these -  now removed  - mutexes were for?

Reply:
originally we were using a planStep to communciate to the obstacle checker how far through the plan we were. This allowed us to only check the remaining plan. Based on my work with Peter we found the more optional solution to be editing the plan's starting position.

Reply:
Note the current implementation of this only works on map_typ=none where any point chosen will result in a two step plan. When SLAM/GPS explore comes into the picture, we will need to revise this

Reply:
I assume we want to limit the scope fo this project to not include those cases but let me know if that is a mistaken assumption

Reply:
Yes - only map type none for this iteration. Add a TODO with a ticket, I gave a feeling implementing this for other map types will change this code heavily anyway.